src/main.cpp

```diff
@@ -1003,6 +974,13 @@ bool AcceptToMemoryPool(CTxMemPool& pool, CValidationState &state, const CTransa
         view.SetBackend(dummy);
         }
 
+        CAmount nValueOut = tx.GetValueOut();
+        CAmount nFees = nValueIn-nValueOut;
+        // Don't accept it if it can't get into a block
+        if (nFees < 0)
+            return state.DoS(0, error("%s: negative fees %s, %d", __func__, hash.ToString(), nFees),
+                             REJECT_INSUFFICIENTFEE, "insufficient fee");
```

Comment:
Probably this should be replaced with bad-txns-fee-negative to better keep track of the validation duplication.